src/hotspot/cpu/x86/compressedKlass_x86.cpp

```diff
@@ -40,9 +41,12 @@ char* CompressedKlassPointers::reserve_address_space_for_compressed_classes(size
     if (result == nullptr) {
       result = reserve_address_space_for_zerobased_encoding(size, aslr);
     }
-  } // end: low-address reservation
+  } else {
```

Comment:
This should be:

    if (result == nullptr) {
        // If we cannot use zero-based encoding (when CDS is enabled), optimizing for an
        ...
    }

Reply:
I don't think so. Why?

Reply:
I think the point is to reflow this into the series of ifs:

```
char* result = nullptr;
if (optimize_for_zero_base) {
   result = ...
}

if (result == nullptr) {
   result = ...
}
```

Probably good for future maintenance if we want to add more cases in this block.

Reply:
Ah okay. But that's wrong, since for the optimize_for_zero_base case, we would search the low address regions twice.

Reply:
Why do you need to whole space to lie between 0 and 4G then, it seems you only want to have the base being < 4G, then the top of the klass space can be upto size + 4G

Reply:
Ah, right, good catch.